File: packages/java/tests/gradle/kotlin-gradle-test/src/main/kotlin/com/vaadin/hilla/gradle/test/data/Person.kt
package com.vaadin.hilla.gradle.test.data

import jakarta.persistence.Entity
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import java.time.LocalDate

@Entity
class Person(
    var firstName: @NotBlank String,
    var lastName: String = "",
    var email: @Email String? = null,
    var phone: String = "",
    var dateOfBirth: LocalDate? = null,
    var occupation: String? = null,
    var role: String? = null,
    var important: Boolean = false,
): AbstractEntity()

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/kotlin/basic/snapshots/com/vaadin/hilla/parser/plugins/nonnull/kotlin/basic/data/Person.ts
import type AbstractEntity_1 from "./AbstractEntity.js";
import type Address_1 from "./Address.js";
import type Employee_1 from "./Employee.js";
interface Person extends AbstractEntity_1 {
    firstName: string;
    lastName: string;
    email?: string;
    phone: string;
    address: Array<Address_1>;
    team: Record<string, Array<Employee_1 | undefined>>;
    dateOfBirth?: string;
    occupation?: string;
    role?: string;
    important: boolean;
}
export default Person;
